erts/preloaded/src/erlang.erl

```diff
@@ -2197,8 +2197,11 @@ get_module_info(_Module, _Item) ->
     erlang:nif_error(undefined).
 
 %% Shadowed by erl_bif_types: erlang:hd/1
--spec hd(List) -> term() when
-      List :: [term(), ...].
+-spec hd(List) -> Head when
+      List :: nonempty_maybe_improper_list(T, Last),
+      Head :: term(),
+      T :: term(),
+      Last :: term().
```

Comment:
I don't find the extra variables helpful.

```suggestion
-spec hd(List) -> Head when
      List :: nonempty_maybe_improper_list(),
      Head :: term().
```

Reply:
done